docs/en/api/S3-API.md

```diff
@@ -18,375 +18,1701 @@ sharing with file system based applications, and storage system abstraction (e.g
 
 <table class="table table-striped">
-  <tr><th>S3 Feature</th><th>Status</th></tr>
-{% for item in site.data.table.s3-api-supported-operations %}
   <tr>
-    <td>{{ item.S3Feature }}</td>
-    <td>{{ item.Status }}</td>
+    <th>S3 API Action</th>
+    <th>REST API Endpoint</th>
+    <th>Supported Headers</th>
+    <th>Supported Query Parameters</th>
+  </tr>
+{% for item in site.data.table.s3-api-supported-actions %}
+  <tr>
+    <td><a href="https://docs.aws.amazon.com/AmazonS3/latest/API/API_{{ item.action }}.html">{{ item.action }}</a></td>
+    <td>{{ item.endpoint }}</td>
+    <td>
+      {% assign headers = item.headers | split: "|" %}
+      {% if headers.size == 0 %}
+      N/A
+      {% else %}
+      <ul style="list-style-type:none;margin:0;padding:0">
+      {% for header in headers %}
+      {% if forloop.last %}
+      <li>{{ header }}</li>
+      {% else %}
+      <li>{{ header }},</li>
+      {% endif %}
+      {% endfor %}
+      </ul>
+      {% endif %}
+    </td>
+    <td>
+      {% assign params = item.queryParams | split: "|" %}
+      {% if params.size == 0 %}
+      N/A
+      {% else %}
+      <ul style="list-style-type:none;margin:0;padding:0">
+      {% for param in params %}
+      {% if forloop.last %}
+      <li>{{ param }}</li>
+      {% else %}
+      <li>{{ param }},</li>
+      {% endif %}
+      {% endfor %}
+      </ul>
+      {% endif %}
+    </td>
   </tr>
 {% endfor %}
 </table>
 
-### Limitation
-In Alluxio, we use `/` as a reserved separator. Therefore, any S3 directory with an object named `/` (eg: `s3://example-bucket//`) will conflict and behave incorrectly.
+## Property Keys
 
-### Bucket
-Bucket must be a directory directly under a mount point.   
-If it is under a non-root mount point, the bucket separator must be used as the separator in the bucket name.   
-For example, mount:point:bucket represents Alluxio directory /mount/point/bucket.  
+The following table contains the configurable
+[Alluxio property keys]({{ '/en/reference/Properties-List.html' | relativize_url }})
+which pertain to the Alluxio S3 API.
 
-## Language support
-Alluxio S3 client supports various programming languages, such as C++, Java, Python, Golang, and Ruby.
-In this documentation, we use curl REST calls and python S3 client as usage examples.
+<table class="table table-striped">
+<tr><th>Property Name</th><th>Default</th><th>Description</th></tr>
+{% for item in site.data.table.common-configuration %}
+  {% if item.propertyName contains "alluxio.proxy.s3" %}
+  <tr>
+    <td><a class="anchor" name="{{ item.propertyName }}"></a> {{ item.propertyName }}</td>
+    <td>{{ item.defaultValue }}</td>
+    <td>{{ site.data.table.en.common-configuration[item.propertyName] }}</td>
+  </tr>
+  {% endif %}
+{% endfor %}
+</table>
 
 ## Example Usage
 
-### REST API
-For example, you can run the following RESTful API calls to an Alluxio cluster running on localhost.
-The Alluxio proxy is listening at port 39999 by default.
```

Comment:
The entire `REST API` examples section was collapsed into `navtabs` alongside `aws` cli examples.

They were mapped to be one-to-one with their corresponding S3 API Actions.